### DISCUSSION TITLE
I believe I have a False positive when annotating function argument with module extension in type

### DISCUSSION THREAD

**Discussion by mazborowski** [microsoft/pyright#9045] [Q&A]
Using python3.12, if I have the following:

```
import typing as t
def function(a: t.Any): pass
```
I will get `Illegal type annotation: variable not allowed unless it is a type alias (PyRight reportGeneralTypeIssues)`

If, however, I were to import `Any` from typing and use that as an annotation, I do not get an error.
```
from typing import Any
def function(a: Any): pass  # okay
```

I have never had a problem with this in the past. Recently I did update my OS (and Python along with it), and this would be my first time using PyRight since.

Can anyone confirm that this is correct behavior, please?

**Comment by erictraut** [comment]
I'm not able to repro this issue with a recent version of pyright. 

Code sample in [pyright playground](https://pyright-play.net/?code=JYWwDg9gTgLgBDAnmYA7A5nAhgZwQKHwBMBTAMzjIFdUBjGYCVACiwC4EA6AQVUQEo2%2BOCLhhcOfEA)

Please verify your pyright version is up to date. If so, are there any additional steps required to repro the behavior you're seeing?

**Reply by mazborowski** [reply]
I can't repro this anywhere else.  It's only one file so far that this is happening.

![image](https://github.com/user-attachments/assets/f223673c-5f1e-4f0c-98ec-b0f02e14097a)

I am using coc-pyright@1.1.380

**Reply by erictraut** [reply]
If you can provide repro steps or a minimal self-contained code sample, I'm happy to investigate further. 

I suspect this is related to a bug that I recently fixed in 1.1.381. Please try upgrading to the latest version and see if that fixes it.

**Comment by mazborowski** [comment]
Thanks for your time. Turns out I had a variable `t` declared later in my file. After I changed its name, the error went away.

Both 1.1.380 and the current HEAD of main will show what I saw:
```
import typing as t

def foo(bar: t.Optional[str]): pass -> None
                                                                                                                          
t = '123'
```

**Comment by mazborowski** [comment]
Your bug fix lives to fight another battle!
